### PR TITLE
ci: Adding composite action to setup the project

### PR DESCRIPTION
**Pull request by Tohaker**
Small improvements to the workflows to make them more DRY and repeatable

### CODE REVIEW

File: .github/actions/setup-project/action.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    - uses: actions/setup-node@v3
+      with:
+        node-version: ${{ inputs.node-version }}
+        cache: "pnpm"
+    - name: Install Turbo
+      run: npm i -g turbo@${{ inputs.turbo-version }}
+      shell: bash
+    - name: Install dependencies
+      run: pnpm install
+      shell: bash
```
Comment (danstarns) [inline .github/actions/setup-project/action.yml:29]: Do we need to specify that is bash for each step?

Comment (Tohaker) [reply .github/actions/setup-project/action.yml:29]: Yes, it's required in composite actions

https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#runsstepsshell

**Review by danstarns — APPROVED** [review]
This is great, It splits up the linting and testing and only installs puppeteer on the tests. 

This will help the project expand its CI and testing strategies. 

Thanks @Tohaker 🤘

File: .github/actions/setup-project/action.yml (hunk 1 above)
Comment (Tohaker) [reply .github/actions/setup-project/action.yml:29]: We can choose a different shell if we need, but bash should do for now

**Comment by danstarns** [comment]
Going to do a release now with the changes to the pipeline to ensure all is working correctly.

File: .github/actions/setup-project/action.yml (hunk 1 above)
Comment (danstarns) [reply .github/actions/setup-project/action.yml:29]: Thank you for the clarification.

**Comment by danstarns** [comment]
alpha.47 was released successfully. Thank you again 🙏

**Comment by danstarns** [comment]
https://github.com/rocket-connect/graphql-debugger/actions/runs/6151366315/job/16691261318

**Pull request merged**